### PR TITLE
[FIVE-188] Realizar UnitTest para el controlador de Artefactos

### PR DESCRIPTION
**Pull request by gperelstein**
Se debe crear la clase de UnitTest para el controlador de Artefactos

Cambios realizados:

- Se creó la clase ArtifactControllerTests con los tests correspondientes al ArtifactsController

- Se modifico el modelo de Artifact en FRF.Core.Models para que la fecha de modificación pueda ser nula

### CODE REVIEW

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+            var returnValueList = returnValue.ToList();
+
+            AssertCompareList(artifacts, returnValueList);
+            
+            _artifactsService.Verify(mock => mock.GetAll(), Times.Once);
+            _mapper.Verify(mock => mock.Map<IEnumerable<ArtifactDTO>>(It.Is<List<Artifact>>(a => a.Contains(artifact))), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetAllByProjectIdAsync()
```
Comment (matiasbeckerle) [inline test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:125]: _ReturnsOk

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+        public async Task GetAsync_ReturnsNotFound()
+        {
+            // Arrange
+            var artifactId = 1;
+
+            //Act
+            var result = await _classUnderTest.GetAsync(artifactId);
+
+            // Assert
+            var notFoundResult = Assert.IsType<NotFoundResult>(result);
```
Comment (matiasbeckerle) [inline test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:309]: If you're not gonna use the object, you can just call `Assert.IsType<NotFoundResult>(result);`

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,637 @@
...
+
+            var artifactDTO = new ArtifactDTO
+            {
+                Id = artifact.Id,
+                Name = artifact.Name,
+                Provider = artifact.Provider,
+                Settings = artifact.Settings,
+                ProjectId = projectId,
+                ArtifactType = artifactTypeDTO
+            };
```
Comment (matiasbeckerle) [inline test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: Food for thought: there are some alternatives where you can make the _arrange_ of a test method smaller or shorter. Have that in mind. Start to see patterns. You will likely revisit tests later with newer ideas. Don't worry, they will flow naturally. Nothing to change now.

**Review by matiasbeckerle — APPROVED** [review]

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs (hunk 1 above)
Comment (gperelstein) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs]: Done

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs (hunk 2 above)
Comment (gperelstein) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs]: Done

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs (hunk 3 above)
Comment (gperelstein) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: I was hoping to have time to see this before the next demo but it was not possible. But when I was creating this test thought that I was repeating the objects in the arrange of the tests. I thought to define them as a property of the class so I can use them in all of the tests. Is this correct?

Comment (matiasbeckerle) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: It's an option. Not bad. You can start with that and see how it evolves. The drawback of the approach is when each test needs (in order to simulate the different scenarios) particular values. But I'm getting ahead of time. Do it and we can review later on if needed.

**Review by matiasbeckerle — APPROVED** [review]

File: test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs (hunk 3 above)
Comment (gperelstein) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: If it's ok I merge this test like this and I create a task to improve this in the next sprint so I can close this ticket for now while I'm working on another ticket. It's ok?

Comment (matiasbeckerle) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: Of course!

Comment (gperelstein) [reply test/FRF.Web.Tests/Controllers/ArtifactsControllerTests.cs:90]: Perfect!

**Pull request merged**